### DISCUSSION TITLE
Setting multiline content generates |-

### DISCUSSION THREAD

**Discussion by AnderssonPeter** [mikefarah/yq#1384] [Q&A] [answered]
Hi I'm using yq 4.28.1 the following way

`cloud_init_content=$(./yq ".ssh_keys.${key_type}_private = \"$key_private\"" <<< $cloud_init_content)` where `$key_private` is a openssh key so something like
```
-----BEGIN OPENSSH PRIVATE KEY-----
base64 string with
linebreaks
-----END OPENSSH PRIVATE KEY-----
```

But the output yml has 
```
ssh_keys:
  rsa_private: |-
    -----BEGIN OPENSSH PRIVATE KEY-----
    .....
```

The problem here is that cloud-init doesn't like the - after the | is there any way to stop yq from generating it?
So what I'm after is ` rsa_private: |` and not ` rsa_private: |-`.

**Comment by AnderssonPeter** [comment]
Found a temporary workaround `config_content="${config_content//: |-$'\n'/: |$'\n'}"` but I have never seen |- before so I'm not 100% sure if its a bug or not

**Comment by mikefarah** [comment, marked as answer]
Gday - take a look at this documentation here: https://mikefarah.gitbook.io/yq/operators/string-operators#string-blocks-bash-and-newlines

Basically the dash at the end of `|-` indicates that there is _no_ new line at the end of the string (https://yaml-multiline.info/)
